python/dgl/dataloading/dataloader.py

```diff
@@ -336,8 +339,41 @@ def __len__(self):
         ) // self.batch_size
 
 
+def _numel_of_shape(shape):
+    return reduce(operator.mul, shape, 1)
+
+
+def _init_gpu_caches(graph, gpu_caches):
+    if not hasattr(graph, "_gpu_caches"):
+        graph._gpu_caches = {}, {}
+    if gpu_caches is None:
+        return
+    if not isinstance(gpu_caches, tuple):
+        gpu_caches = gpu_caches, {}
+    for i, frames in enumerate([graph._node_frames, graph._edge_frames]):
+        for tid, frame in enumerate(frames):
+            type_ = [graph.ntypes, graph.canonical_etypes][i][tid]
```

Comment:
why naming a variable with suffix `_`?  In convention it indicates an internal member variable.

Reply:
I am following the variable naming convention in other parts of the code in this file.

Reply:
I see, so that should be fine.